rust/lance/src/index/vector/ivf.rs

```diff
@@ -1382,6 +1383,65 @@ fn generate_remap_tasks(offsets: &[usize], lengths: &[u32]) -> Result<Vec<RemapP
     Ok(tasks)
 }
 
+#[allow(clippy::too_many_arguments)]
+pub(crate) async fn remap_index_file_v3(
+    dataset: &Dataset,
+    new_uuid: &str,
+    index: Arc<dyn VectorIndex>,
+    mapping: &HashMap<u64, Option<u64>>,
+    column: String,
+) -> Result<()> {
+    let index_dir = dataset.indices_dir().child(new_uuid);
+    let element_type = get_vector_element_type(dataset, &column)?;
+    match index.index_type() {
```

Comment:
Would it be possible to add a `remap` method to the `VectorIndex` trait instead of using a match statement here?

Reply:
yeah just tried, it can work!